src/models/tag/dto/tagDto.prisma.ts

```diff
@@ -1,32 +1,75 @@
-import { Prisma, Tag } from '@prisma/client';
+import { Prisma, Tag, Task } from '@prisma/client';
 import { prisma } from '../../../utils/prisma/prisma-client';
+import ITagPayload from '../types/TagPayload.args';
+
+export interface TagDetails extends Tag {
+tasks: Task[],
+}
 
 export default function TagPrismaDto() {
+  //* Create method
+  async function createTag(payload: ITagPayload): Promise<TagDetails | null > {
+    return prisma.tag.create({
+      data: {
+        label: payload.label,
+        color: payload.color,
+      },
+      include: {
+        tasks: true,
+      },
+    });
+  }
+  //* Update
+  async function updateTag(payload: ITagPayload, id : Prisma.TagWhereUniqueInput):
+  Promise<TagDetails | null > {
+    return prisma.tag.update({
+      where: id,
+      data: {
+        label: payload.label,
+        color: payload.color,
+      },
+      include: {
+        tasks: true,
+      },
+    });
+  }
   // ** READ
-  async function all(): Promise<Tag[]> {
-    return prisma.tag.findMany();
+  async function all(): Promise<TagDetails[]> {
+    return prisma.tag.findMany({
+      include: {
+        tasks: true,
+      },
+    });
   }
 
   async function oneById(
     id: Prisma.TagWhereUniqueInput,
-  ): Promise<Tag | null> {
+  ): Promise<TagDetails | null> {
```

Comment:
```suggestion
  ): Promise<TagDetails> {
```
Pourquoi mettre en type `Promise<TagDetails | null>` ici ?